js/script.js

```diff
@@ -369,11 +386,39 @@ const renderTodos = () => {
 
     activateFilterButton(todos, filterState);
 
+    updateLocalStorageData(searchedValue);
+
     paginatedTodos.forEach((todo) => {
         $todoList.appendChild(createTodoElement(todo));
     });
 };
 
+const getLocalStorageData = () => {
+    const data = JSON.parse(localStorage.getItem("todos"));
+    const state = JSON.parse(localStorage.getItem("state"));
```

Comment:
`JSON.parse`, `JSON.stringfy` are prone to break. Move them to a separate file and use `try..catch` block.

```js
// dbOperations.js

export function getTodos() {
  const payload = {
    todos: [],
    error: null
  }
  try {
    payload.todos = JSON.parse(localStorage.getItem("todos"))
  } catch (error) {
    payload.error = error
  } finally {
    return payload
  }
}
```

**Usage:**
```js
const {todos, error} = getTodos()

if(error){
  // handle error here by showing toast notification
  return
}

const {state, error} = getState()

if(error){
  return
}

// Do work with todos and state
```